src/me/mrletsplay/shittyauthlauncher/ShittyAuthSettingsController.java

```diff
@@ -17,7 +46,83 @@ public class ShittyAuthSettingsController {
 	
 	@FXML
 	private CheckBox checkboxMinimizeLauncher;
-	
+
+	@FXML
+	private Label textUsing;
+
+	@FXML
+	private VBox boxMirrors;
+
+	public void init(ShittyAuthController controller) {
+		this.controller = controller;
+		mirrors = FXCollections.observableArrayList(new ArrayList<>());
+
+		mirrors.addListener((ListChangeListener<DownloadsMirror>) v -> {
+			boxMirrors.getChildren().removeIf(c -> !(c instanceof Button));
+			for(DownloadsMirror acc : mirrors) {
+				boxMirrors.getChildren().add(createMirrorItem(acc));
+			}
+		});
+
+		//<Default Mirror>
+		DownloadsMirror mirror = new MojangMirror();
+		mirrors.add(mirror);
+		ShittyAuthLauncher.mirror = mirror;
+		MinecraftVersion.initVersions(mirror);
+		controller.loadVersions();
+		textUsing.setText("Current Mirror: "+mirror.name);
+		//</Default Mirror>
+
+		File mirrorsJSON = new File("shittyauthlauncher/mirrors.json");
+
+
+		List<DownloadsMirror> checkedMirrors = new ArrayList<>();
+		if(mirrorsJSON.exists()) {
+			checkedMirrors = loadMirror(mirrorsJSON);
+		}else {
+			try {
+				JSONObject obj = new JSONObject();
+				obj.put("mirrors", new JSONObject());
+				Files.writeString(mirrorsJSON.toPath(), obj.toFancyString());
+			} catch (IOException e) {
+				System.err.println("Failed to write file");
+				e.printStackTrace();
+			}
+		};
+		mirrors.addAll(checkedMirrors);
+	}
+
+
+	public static List<DownloadsMirror> loadMirror(File mirrorsFile) {
+		JSONObject p;
+		try {
+			p = new JSONObject(Files.readString(mirrorsFile.toPath(), StandardCharsets.UTF_8));
+		} catch (IOException e) {
+			throw new FriendlyException("Failed to read file", e);
+		}
+
+		List<DownloadsMirror> mirrorsToLoad = new ArrayList<>();
+
+		JSONArray pr = p.getJSONArray("mirrors");
+		for(Object loopMirror : pr.toArray()) {
+			JSONObject mirrorJSON = (JSONObject) loopMirror;
+
+			boolean custom = mirrorJSON.getBoolean("custom");
+			String name = mirrorJSON.getString("name");
+			String versionManifest = mirrorJSON.getString("versionManifest");
+			String assetsURL = mirrorJSON.getString("assetsURL");
+
+
+			DownloadsMirror mirrorAdding = new DownloadsMirror();
+			mirrorAdding.custom = custom;
+			mirrorAdding.name = name;
+			mirrorAdding.versionManifest = versionManifest;
+			mirrorAdding.assetsURL = assetsURL;
+			mirrorsToLoad.add(mirrorAdding);
```

Comment:
This could be simplified since `DownloadsMirror` is a `JSONConvertible`. Instead of getting each property manually, just use:
```java
JSONObject mirrorJSON = (JSONObject) loopMirror;
DownloadsMirror mirror = JSONConverter.decodeObject(mirrorJSON, DownloadsMirror.class);
mirrorsToLoad.add(mirror);
```